### ISSUE TITLE
Maven resolver seems to be slower than the Coursier resolver

### ISSUE THREAD

**Issue by tharakadesilva** [bazelbuild/rules_jvm_external#1153]
Our Renovate pipeline that takes 10 branches and runs the repinning is taking longer than 1 hour to finish (Coursier used to finish in ~8-12 minutes).

I have noticed that after all the deps have been downloaded, it just pauses for a considerable amount of time with no logs where I am assuming it is running some sort of resolving process / writing to the lock files.

Not sure what metrics I can help gather, but please let me know if I can gather any additional information for you.

**Comment by shs96c** [comment]
You can get additional logging by running the resolution with the adding `RJE_VERBOSE=1` to the environment variables. If the information this provides doesn't help much, I can add some additional logging to the next `rules_jvm_external` release.

There are also some other [environment variables you might want to play with](https://github.com/bazelbuild/rules_jvm_external?tab=readme-ov-file#configuring-maven).

**Comment by tharakadesilva** [comment]
@shs96c I got some time to dig into this today and it turned out to be an issue in our configuration.

We didn't have [execution mode](https://docs.renovatebot.com/configuration-options/#executionmode) set, so it was running `REPIN=1 bazel run @maven://pin` on every item on the change request.

So, if we had one PR for AWS deps with like 30 deps, it would try to run it 30 times...

I am not sure how Coursier survived this, but I guess it had some sort of caching in place.

---

Thanks to your suggestion, I also started using `RJE_MAX_THREADS` and `RJE_UNSAFE_CACHE`. `RJE_ASSUME_PRESENT` didn't really work and I didn't get time to look into it in depth.

---

Apologies for the trouble and thanks for the tips!!
